src/miniscript/iter.rs

```diff
@@ -98,43 +94,93 @@ impl<Pk: MiniscriptKey, Ctx: ScriptContext> Miniscript<Pk, Ctx> {
     /// Otherwise returns an empty `Vec`.
     ///
     /// For each public key the function computes hash; for each hash of the public key the function
-    /// returns it's cloned copy.
+    /// returns it cloned copy.
     ///
     /// NB: The function analyzes only single miniscript item and not any of its descendants in AST.
     /// To obtain a list of all public key hashes within AST use [`iter_pkh()`] function,
     /// for example `miniscript.iter_pubkey_hashes().collect()`.
     pub fn get_leaf_pkh(&self) -> Vec<Pk::Hash> {
-        match self.node.clone() {
-            Terminal::PkH(hash) => vec![hash],
-            Terminal::PkK(key) => vec![key.to_pubkeyhash()],
-            Terminal::Multi(_, keys) => keys.iter().map(Pk::to_pubkeyhash).collect(),
+        match self.node {
+            Terminal::PkH(ref hash) => vec![hash.clone()],
```

Comment:
Hashes are `Copy`, no need for `ref` or `clone`

Reply:
Hases are, but not `MiniscriptKey::Hash`. This does not compile, I either have to leave it as is, or add `Hash: Copy` type in `MiniscriptKey` trait